views/js/controller/runtime/testRunner.js

```diff
@@ -256,7 +265,36 @@ define(['jquery', 'lodash', 'spin', 'serviceApi/ServiceApi', 'serviceApi/UserInf
 			    }
 			}
 		},
+        /**
+         * Mark apropriate timer by warning colors and show feedback message
+         * @param {object} cst - Time constraint
+         * @returns {undefined}
+         */
+        timeWarning : function (cst) {
+            $('#qti-timers > .qti-timer__type-' + cst.qtiClassName).addClass('qti-timer__warning');
+
+            // Initial time more than warning time in config
+            if (cst.seconds > cst.warningTime) {
+                var hours = Math.floor(cst.warningTime / 3600),
+                    minutes,
+                    seconds,
+                    message = '';
+
+                cst.warningTime = cst.warningTime - hours * 3600;
+                minutes = Math.floor(cst.warningTime / 60);
+                seconds = cst.warningTime - minutes * 60;
+                message = '';
+
+                if (hours) message += hours + ' ' + __(hours === 1 ? 'hour' : 'hours') + ' ';
+                message += minutes + ' ' + __(minutes === 1 ? 'minute' : 'minutes');
+                if (seconds) message += ' ' + seconds + ' ' + __(seconds === 1 ? 'second' : 'seconds');
```

Comment:
I don't know if it can help you but the [moment](http://momentjs.com/) library is available